src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts

```diff
@@ -273,11 +275,20 @@ export class OpenQueryAction extends Action {
 	public static ID = 'ep.OpenQueryAction';
 	public static LABEL = localize('openQueryAction', "Open Query");
 
-	constructor() {
+	constructor(private triggeredFromContextMenu: boolean,
+		@IAdsTelemetryService private readonly telemetryService: IAdsTelemetryService
+	) {
 		super(OpenQueryAction.ID, OpenQueryAction.LABEL, openQueryIconClassNames);
 	}
 
 	public override async run(context: ExecutionPlanView): Promise<void> {
+		let source = this.triggeredFromContextMenu ? 'Context menu' : 'Action bar';
```

Comment:
please define a type for the source.

`export type ExecutionPlanActionSource = 'ContextMenu'|'ActionBar';`

Reply:
Do I export the new type from here or a different file?

Reply:
in this file should be fine. not a huge deal.